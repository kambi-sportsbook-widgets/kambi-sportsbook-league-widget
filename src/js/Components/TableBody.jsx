import React from 'react';

const TableBody = ({ children }) => {
   return (
      <main className="KambiWidget-font l-flexbox l-vertical l-flexed l-pack-start">
         {children}
      </main>
   );
};

TableBody.propTypes = {
   /**
    * Row components
    */
   children: React.PropTypes.arrayOf(React.PropTypes.element).isRequired
};

export default TableBody;
