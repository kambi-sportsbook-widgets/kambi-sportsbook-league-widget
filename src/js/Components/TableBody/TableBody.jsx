import React, { Children, PropTypes } from 'react';
import styles from './TableBody.scss';

const TableBody = ({ children }) => {
   // todo: use TableBodyRow (KSBWI-908)
   if (!children instanceof Array) {
      children = [];
   }

   return (
      <tbody>
         {children.map((row, i) =>
            <tr key={i} className={['KambiWidget-card-inner-border', styles.row].join(' ')}>
               {row}
               <td className='margin' />
            </tr>
         )}
      </tbody>
   );
};

TableBody.propTypes = {
   /**
    * Inner rows
    */
   children: PropTypes.node
};

export default TableBody;
