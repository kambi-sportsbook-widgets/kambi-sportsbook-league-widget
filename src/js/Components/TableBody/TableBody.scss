:local(.row) {
   border-bottom-style: solid;
   border-bottom-width: 1px;
   height: 46px;

   td.margin {
      width: 16px;
   }
}
