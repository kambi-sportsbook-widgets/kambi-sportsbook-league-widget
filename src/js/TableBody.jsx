import React from 'react';
import { translationModule } from 'widget-core-library';
import TableRow from './TableRow';

const TableBody = ({ participants, columns, event, updated }) => {
   const t = translationModule.getTranslation.bind(translationModule);

   return (
      <main className="KambiWidget-font l-flexbox l-vertical l-flexed l-pack-start">
         {participants.map(participant => <TableRow key={participant.participantId} participant={participant} columns={columns} event={event} />)}
         <div className="kw-table-item l-flexbox l-align-center l-pl-16 l-pr-16 l-pt-6 l-pb-6">{t('Last updated')}: {updated}</div>
      </main>
   );
};

TableBody.propTypes = {
   participants: React.PropTypes.array.isRequired,
   columns: React.PropTypes.array.isRequired,
   event: React.PropTypes.object,
   updated: React.PropTypes.string
};

export default TableBody;
