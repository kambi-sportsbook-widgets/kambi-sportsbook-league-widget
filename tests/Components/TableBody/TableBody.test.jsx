/* eslint-env jest */
import React, { Children } from 'react';
import TableBody from '../../../src/js/Components/TableBody/TableBody';
import ReactTestUtils from 'react-addons-test-utils';

const renderer = ReactTestUtils.createRenderer();

describe('TableBody DOM rendering', () => {

   it('renders correctly without children', () => {
      expect(renderer.render(
         <TableBody />
      )).toMatchSnapshot();
   });

   it('renders correctly with children', () => {
      expect(renderer.render(
         <TableBody>
            <tr><td/><td/></tr>
            <tr><td/><td/></tr>
         </TableBody>
      )).toMatchSnapshot();
   });

});
