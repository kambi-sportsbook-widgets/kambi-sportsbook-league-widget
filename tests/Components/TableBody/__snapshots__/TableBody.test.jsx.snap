// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody DOM rendering renders correctly with children 1`] = `
<tbody>
  <tr
    className="KambiWidget-card-inner-border row"
  >
    <tr>
      <td />
      <td />
    </tr>
    <td
      className="margin"
    />
  </tr>
  <tr
    className="KambiWidget-card-inner-border row"
  >
    <tr>
      <td />
      <td />
    </tr>
    <td
      className="margin"
    />
  </tr>
</tbody>
`;

exports[`TableBody DOM rendering renders correctly without children 1`] = `
<tbody>
  
</tbody>
`;
